regexp.go

```diff
@@ -175,6 +173,18 @@ func (r *routeRegexp) url(values map[string]string) (string, error) {
 	return rv, nil
 }
 
+// getUrlQuery returns a single query parameter from a request URL.
+// For a URL with foo=bar&baz=ding, we return only the relevant key
+// value pair for the routeRegexp.
+func (r *routeRegexp) getUrlQuery(req *http.Request) string {
+	keyVal := strings.Split(r.template, "=")
+	if len(keyVal) == 0 {
+		return ""
+	}
+	re := regexp.MustCompile(keyVal[0] + "[^&]*")
```

Comment:
I don't like the idea of compiling a regexp during every match attempt, that's going to hurt performance a lot.